packages/http/src/validator/validators/body.ts

```diff
@@ -86,20 +88,30 @@ function deserializeAndValidate(content: IMediaTypeContent, schema: JSONSchema,
   );
 }
 
+/**
+ * Given a schema, return an equivalent schema that does not include `allOf`.
+ */
+function withoutAllOf(s: JSONSchema): JSONSchema {
+  return mergeAllOf(s, {
+    ignoreAdditionalProperties: true,
+    deep: true,
+  });
+}
+
 function memoizeSchemaNormalizer(normalizer: SchemaNormalizer): SchemaNormalizer {
   const cache = new WeakMap<JSONSchema7, O.Option<JSONSchema7>>();
   return (schema: JSONSchema7) => {
     const cached = cache.get(schema);
     if (!cached) {
-      const newSchema = normalizer(schema);
+      const newSchema = normalizer(withoutAllOf(schema));
```

Comment:
@rainum , I made this change here rather than in [`stripReadOnlyProperties()](https://github.com/stoplightio/prism/blob/8888233a91415284de9151f5cb4e79dc98248628/packages/http/src/utils/filterRequiredProperties.ts#L91) because I think we want symmetrical behavior for `writeOnly` properties when validating output.  Is that correct?

Reply:
> I think we want symmetrical behavior for writeOnly properties when validating output.

Perfect 👌 You're completely right